Competition/src/main/cpp/Arm.cpp

```diff
@@ -0,0 +1,98 @@
+#include "Arm.h"
+
+Arm::Arm() : armMtrLeft{ArmConstants::TALON_ID_LEFT_ARM}, 
+             armMtrRight{ArmConstants::TALON_ID_RIGHT_ARM} {
+
+}
+
+Arm::Arm(frc::XboxController* controller) : operator_controller(controller), 
+                                            armMtrLeft{ArmConstants::TALON_ID_LEFT_ARM}, 
+                                            armMtrRight{ArmConstants::TALON_ID_RIGHT_ARM} {
+    InitArm();
+}
+
+Arm& Arm::GetInstance() {
+    static Arm instance;
+    return instance;
+}
+
+void Arm::Periodic() {
+    assessInputs();
+    assignOutputs();
+}
+
+void Arm::InitArm() {
+    armMtrLeft.ConfigFactoryDefault();
+    armMtrRight.ConfigFactoryDefault();
+
+    armMtrLeft.SetNeutralMode(NeutralMode::Brake);
+    armMtrRight.SetNeutralMode(NeutralMode::Brake);
+
+    armMtrLeft.SetInverted(false);
+    armMtrRight.SetInverted(false);
+
+    state.disengage = true;
+    state.step1_start_time = -1;
+    state.step2_start_time = -1;
+}
+
+void Arm::setDefaultState() {
+    state.arm_state = ArmState::DISABLED;
+}
+
+void Arm::assessInputs() {
+    if (state.disengage) {
+        state.arm_state = ArmState::DISENGAGE;
+    }
+    else if (std::abs(operator_controller->GetY(frc::GenericHID::kLeftHand)) > 0.05) {
+        state.arm_state = ArmState::MANUAL;
+    }
+}
+
+void Arm::assignOutputs() {
+    if (state.arm_state == ArmState::DISENGAGE) {
+        state.curr_time = state.timer.GetFPGATimestamp();
+
+        if (state.step1_start_time == -1) {
+            state.step1_start_time = state.curr_time;
+        }
+        else if (state.curr_time - state.step1_start_time <= 0.2) {
+            state.current_power = 0.4;
+        }
+        else if (state.step2_start_time == -1) {
+            state.step2_start_time = state.curr_time;
+        }
+        else if (state.curr_time - state.step2_start_time <= 0.3) {
+            state.current_power = 0.042;
+        }
+        else {
+            state.current_power = 0;
+
+            // reset code
+            // should reset code be in setDefaultState()?
+            state.disengage = false;
+            state.step1_start_time = -1;
+            state.step2_start_time = -1;
+        }
+    }
+    else if (state.arm_state == ArmState::MANUAL) {
+        if (std::abs(operator_controller->GetY(frc::GenericHID::kLeftHand)) <= 0.05) {
```

Comment:
Reading from the joystick should occur in assessInputs and store into a state.joystick variable.

All the following if statements should then reference state.joystick